### PR TITLE
docs: add auth code with PKCE flow guide

### PR DESCRIPTION
**Pull request by embbnux**

### CODE REVIEW

File: docs/authentication/auth-code-pkce-flow.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+_drLS7o5FwkfUiBhlq2hwJnK_SC6yE7sKOde5O1fdzk
+```
+
+## Step 2. Request authorization code
+
+When your application needs to access a user's data, redirect the user to the RingCentral API server. The authorization URL is same as URL from [Authorization Code Flow Step 1](./auth_code_flow#step-1-request-authorization-code), and PKCE flow will need additional parameters `code_challenge` and `code_challenge_method` in authorization URL:
+
+| Parameter       | Type   | Description |
+| --------------- | ------ | ----------- |
+| `code_challenge` | string | Required. Generated from code challenge
```
Comment (CoCraig) [inline docs/authentication/auth-code-pkce-flow.md:44]: "code_challenge string Required. Generated from code challenge <- this should say code verifier

**Review by CoCraig — COMMENTED** [review]
Make sure to add mkdocs.yml changes.  I made changes to my checked out version so I could review in localhost with rendering.

File: docs/authentication/auth-code-pkce-flow.md (hunk 1 above)
Comment (embbnux) [reply docs/authentication/auth-code-pkce-flow.md:44]: It should be `code_challenge`. We send `code_challenge` to server by authorization URI. And send `code_verifier`  to server only  in token exchange api.

File: docs/authentication/auth-code-pkce-flow.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+**Example Login URL**
+
+Below is an example login URL to initiate the PKCE authorization flow. We recommend developers use an SDK to generate this URL in a more automated fashion.
+
+```
+https://platform.ringcentral.com/restapi/oauth/authorize?response_type=code&redirect_uri=<my_uri>&client_id=<client_id>&display=&prompt=&&code_challenge=<code_challenge_string>&code_challenge_method=S256
+```
+
+## Step 3. User login and consent
```
[the remark this answers is not in the document]
Comment (embbnux) [reply docs/authentication/auth-code-pkce-flow.md:55]: Yeah, we have example and details in Step4 for redirect uri handling

File: docs/authentication/auth-code-pkce-flow.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
+# Authorization Code Flow with Proof Key for Code Exchange
```
[the remark this answers is not in the document]
Comment (embbnux) [reply docs/authentication/auth-code-pkce-flow.md:1]: added

File: docs/authentication/auth-code-pkce-flow.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+## Step 1. Generate code verifier and code challenge
+
+Before we open an authorization request to RingCentral API server, we need to generate a pair of code verifier and code challenge.
+
+Code verifier should be a cryptographically-random string, without '+', '/', and '=' characters.
+
+**Example Code Verifier**
+
+a cryptographically-random example string:
```
[the remark this answers is not in the document]
Comment (embbnux) [reply docs/authentication/auth-code-pkce-flow.md:39]: added

File: docs/authentication/auth-code-pkce-flow.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+```
+pIUgx4tiqFpaOUz0HMc_QbIyQlL901w8mRmkrmhEJ_E
+```
+
+Code challenge is generated from code challenge. For devices that can perform a SHA256 hash, the code challenge is a BASE64-URL-encoded string of the SHA256 hash of the code verifier. Clients that do not have the ability to perform a SHA256 hash are permitted to use the plain code verifier string as the challenge.
+
+**Example Code Challenge**
+
+Generated from upper code verifier with SHA256 hash and BASE64-URL-encoded:
```
[the remark this answers is not in the document]
Comment (embbnux) [reply docs/authentication/auth-code-pkce-flow.md:66]: added

File: docs/authentication/auth-code-pkce-flow.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+| `state`      | string  | This parameter will be included in response if it was specified in the client authorization request. The value will be copied from the one received from the client |
+	
+If authentication has been passed successfully, your application will get a response similar to the following:
+	
+```http
+HTTP/1.1 302 Found
+Location: https://myapp.example.com/oauth2Callback?code=SplxlOBeZQQYbYS6WxSbIA&state=xyz&expires_in=60
+```
+
+## Step 5. Exchange code for token
```
[the remark this answers is not in the document]
Comment (embbnux) [reply docs/authentication/auth-code-pkce-flow.md:117]: merge step 5 and step 6. And provide some Request and Respond example code in step 5

**Pull request merged**